app/src/main/java/com/github/sdpteam15/polyevents/view/activity/admin/ZoneManagementListActivity.kt

```diff
@@ -41,10 +39,16 @@ class ZoneManagementListActivity : AppCompatActivity() {
         }
 
         recyclerView.adapter = ZoneItemAdapter(zones, openZone)
-        val matcher = Matcher { collection -> collection.orderBy(DatabaseConstant.ZoneConstant.ZONE_NAME.value) }
 
         println(Database.currentDatabase)
-        Database.currentDatabase.zoneDatabase!!.getAllZones(matcher, 50, zones).observe(this) {
+        Database.currentDatabase.zoneDatabase!!.getAllZones(
+            null,
+            50,
+            ObservableList<Zone>().observe(this) {
+                zones.clear(it.sender)
+                zones.addAll(it.value.toList().sortedBy { it.zoneName }, it.sender)
```

Comment:
Couldn't we create a Helper function to sort since you do this in almost all the get list methods ?

Reply:
I have done it because I removed `orderBy` in `Query` but I will redo the inverse

Reply:
I mean, we can remove the orderBy (since sometimes it doesn't work properly) and just use a local sort if you want. But we could just create an helper method that takes the sorting attribute and the list and that returns the sorted list

Reply:
It's already done and I will do the `orderBy` immplementation with the test part